src/collection/Stack.php

```diff
@@ -27,7 +27,7 @@ class Stack extends AbstractList {
 	 * 
 	 * @param array|Iterator $collection
 	 */
-	public function __construct($collection = []) {
+	final public function __construct($collection = []) {
```

Comment:
Could you explain why adding the `final` to prevent this class can be extended by other classes rather than adding the ` * @psalm-consistent-constructor` comment annotation on this class constructor?

Reply:
Given my comment this morning, I found two benefits why you should be able to override the constructor. Given your `BookSet`:

```php
class BookSet extends Set {
	public function __construct(Book[] $collection = []) {
		parent::__construct($collection);
	}

	public function add(Book $book) {
		parent::add($book);
	}
}
```

in both cases you want to provide types, that may already be enough to allow only that type to be the _thing_ of the collection.

Second reason is, this is low-level code. Yes, it should be extendable and yes it would be super weird if I cannot overwrite the constructor o_O

Reply:
fixed